include/swift/AST/DiagnosticsCommon.def

```diff
@@ -54,6 +54,9 @@ NOTE(previous_decldef,none,
 NOTE(brace_stmt_suggest_do,none,
      "did you mean to use a 'do' statement?", ())
 
+NOTE(defer_stmt_at_blockend,none,
+     "defer statement at the end of block is pointless", ())
```

Comment:
Pointless because...? It's important to give a little more explanation here, I think. Also, stylistically, "defer" should be surrounded by single quotes, as shown above with "a 'do' statement".

Reply:
I'd suggest something like:
> 'defer' at the end of its scope is redundant and will execute immediately

Probably best for a follow up PR, but it would be nice to have a fixit that simply removes the `defer` and braces.

Reply:
It seems it'd be simpler just to suggest `do` in place of `defer`.

Reply:
thanks @xwu and @jszumski for the reviews. I improved the note message as you suggested in the following commit.

I thought as well about the fix-it from the beginning but did not have the time to see how to implement it, I am still getting familiar with the code base :). As @xwu, probably the easier/simpler thing would be to suggest a `do`. Will look into it.

Reply:
hi @xwu, @jszumski 
I pushed a 3rd commit that simply adds a fix-it to the diagnostic. The fix-it just suggest to replace the `defer` with a `do` statement.